.travis.yml

```diff
@@ -18,12 +26,14 @@ before_install:
     - popd
     - cmake --version
 install:
-    - sudo pip install -r requirements.txt -r requirements-dev.txt -r plugins/geospatial/requirements.txt -r plugins/metadata_extractor/requirements.txt
-    - sudo pip install -r plugins/celery_jobs/requirements.txt -r plugins/hdfs_assetstore/requirements.txt
+    - mkdir -p "${HOME}/.local/bin"
+    - export PATH="${HOME}/.local/bin:${PATH}"
+    - pip install --user -r requirements.txt -r requirements-dev.txt -r plugins/geospatial/requirements.txt -r plugins/metadata_extractor/requirements.txt
+    - pip install --user -r plugins/celery_jobs/requirements.txt -r plugins/hdfs_assetstore/requirements.txt
     # We have to upgrade six or requiring moto can cause other modules to fail
-    - sudo pip install -U six
-    - sudo python -c "import bcrypt"
-    - npm install -g grunt grunt-cli
+    - pip install --user -U six
```

Comment:
It didn't work without the --user (and the Travis folks told me to use it).

I agree, the bcrypt is unnecessary.

On Mon, Mar 23, 2015 at 11:00 AM, Jonathan Beezley <notifications@github.com
> wrote:

> In .travis.yml
> <https://github.com/girder/girder/pull/758#discussion_r26943979>:
>
> >      # We have to upgrade six or requiring moto can cause other modules to fail
> > -    - sudo pip install -U six
> > -    - sudo python -c "import bcrypt"
> > -    - npm install -g grunt grunt-cli
> > +    - pip install --user -U six
>
> I don't think the --user is strictly necessary because travis builds are
> always done in a userspace virtualenv. Getting rid of the sudo probably
> make the initial import bcrypt below unnecessary as well.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/girder/girder/pull/758/files#r26943979>.
>



-- 
David Manthey
R&D Engineer
Kitware Inc.
(518) 881-4439